projects/perception/heart_anomaly_detection/demo.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2020-2021 OpenDR European Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import numpy as np
+import torch
+import argparse
+from opendr.perception.heart_anomaly_detection import GatedRecurrentUnitLearner, \
+ AttentionNeuralBagOfFeatureLearner, get_AF_dataset
+from opendr.engine.data import Timeseries
+
+
+if __name__ == '__main__':
+    # Select the device for running
+    try:
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
I think this will not raise an exception in any case, but it is ok in any case.

Reply:
I think the idea here was to fallback to cpu if torch.cuda.is_available() fails due to broken cuda installation